internal/utils/extract.go

```diff
@@ -48,14 +51,28 @@ func ExtractTarGz(gzipStream io.Reader, destDir string) ([]string, error) {
 			return nil, err
 		}
 
+		if strings.Contains(header.Name, "..") {
+			return nil, fmt.Errorf("not allowed relative path in tar archive")
+		}
+
 		switch header.Typeflag {
 		case tar.TypeDir:
-			return nil, fmt.Errorf("unexepected dir inside the archive, expected to find only files without any tree structure")
-		case tar.TypeReg:
-			if strings.Contains(header.Name, "..") {
-				return nil, fmt.Errorf("not allowed relative path in tar archive")
+			if artifactType == oci.Plugin || artifactType == oci.Rulesfile {
+				return nil, fmt.Errorf("unexepected dir inside the archive, "+
+					"expected to find only files without any tree structure for %q artifacts", artifactType.String())
+			} else if artifactType == oci.Asset {
+				if dirs > 1 {
```

Comment:
Why do we need this check? Some comments and unit tests for this case would be helpful.